File: inheritance.py
class Human:
    def __init__(self, name, age):
        self.name = name
        self.age = age
        #print("Human Constructor.")
        
    def work(self):
        print("Human ", self.name.title(), "works.")
        
class Doctor(Human):
    def __init__(self, name, age, medical_field):
        super().__init__(name, age)
        self.medical_field = medical_field
        #print("Doctor class constructor.")
        
    def work(self):
        super().work()
        print(self.name.title(), " treats patients.")
    
class Engineer(Human):
    def __init__(self, name, age, field):
        super().__init__(name, age)
        self.field = field
        
class Robot:
    def __init__(self):
        pass
    
    def work(self):
        print("Robot works.")
        
h = Doctor("Tint Shwe", 40, 'Heart')
h.work()

h = Engineer("kyaw swar", 38, 'Construction')
h.work()

h = Robot()
h.work()

File: inheritance.java
class Human
{
    private String name;
    
    public Human(String name)
    {
        this.name = name;
        //System.out.println("Human Constructor.");
        
    }

    public void work()
    {
        System.out.println("Human "+this.name+" works.");
    }

}  

class Doctor extends Human
{
    private String medicalField;
    public Doctor(String name, String medicalField)
    {
        super(name);
        this.medicalField = medicalField;
        //System.out.println("Doctor Constructor!");
    }
    public void work()
    {
        System.out.println("Doctor works "+ " in "+ this.medicalField);
    }
}
class Teacher extends Human
{
    private String school;
    public Teacher(String name, String school)
    {
        super(name);
        this.school = school;

        //System.out.println("Teacher Constructor!");
    }
    public void work()
    {
        System.out.println("Teacher teaches at " + this.school);
    }

}

public class inheritance 
{
    public static void main(String []args)
    {
        Human h = new Doctor("Tint Shwe", "Heart");
        h.work();
        h = new Teacher("Nyein", "BEHS.PG");
        h.work();

    }

}
